gnocchi/rest/__init__.py

```diff
@@ -535,21 +535,41 @@ def MetricSchema(cls, definition):
         return definition
 
     @pecan.expose('json')
-    def post(self):
+    def post(self, source=None):
         creator = pecan.request.auth_helper.get_current_user(
             pecan.request)
         body = deserialize_and_validate(self.MetricSchema)
-        try:
-            m = pecan.request.indexer.create_metric(
-                uuid.uuid4(),
-                creator,
-                name=body.get('name'),
-                unit=body.get('unit'),
-                archive_policy_name=body['archive_policy_name'])
-        except indexer.NoSuchArchivePolicy as e:
-            abort(400, e)
+
+        if not source:
+            try:
+                m = pecan.request.indexer.create_metric(
+                    uuid.uuid4(),
+                    creator,
+                    name=body.get('name'),
+                    unit=body.get('unit'),
+                    archive_policy_name=body['archive_policy_name'])
+            except indexer.NoSuchArchivePolicy as e:
+                abort(400, e)
+        else:
+            source_metric = pecan.request.indexer.list_metrics(
+                id=source, details=True)[0]
```

Comment:
No, list_metrics with `id=` is the good call. It just does a `SELECT` with the right where, so it's equivalent to what would be a `get_metric`.